### ISSUE TITLE
debuerreotype-apt-get hangs with interactive commands when script is disowned

### ISSUE THREAD

**Issue by smijolovic** [debuerreotype/debuerreotype#98]
executing "debuerreotype-apt-get buster dist-upgrade -yqq" runs normally when a script is running with a tty...however when the script is disowned (& disown)...it just hangs.

debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 6460 files and directories currently installed.)
Preparing to unpack .../tzdata_2020d-0+deb10u1_all.deb ...
Unpacking tzdata (2020d-0+deb10u1) over (2020a-0+deb10u1) ...
Setting up tzdata (2020d-0+deb10u1) ...

Current default time zone: 'Etc/UTC'
Local time is now:      Sat Nov  7 00:03:08 UTC 2020.
Universal Time is now:  Sat Nov  7 00:03:08 UTC 2020.
Run 'dpkg-reconfigure tzdata' if you wish to change it.

I even added the following to the script:
export TERM=linux
export DEBIAN_FRONTEND=noninteractive
export APT_LISTCHANGES_FRONTEND=none

Still continues to hang.

Same issue with:
debuerreotype-apt-get buster install -yqq --no-install-recommends iputils-ping iproute2

Any invocation of debuerreotype-apt-get with a non-interactive response works when disowned.  For example:
debuerreotype-apt-get buster update -qq

**Comment by tianon** [comment]
Am I understanding correctly that you're running something like `debuerreotype-apt-get buster dist-upgrade -yqq &` and that's what's causing the problem?  Is there a particular reason you're trying to background it?

**Comment by smijolovic** [comment]
Yes - it's part of a much larger automated build to create a buster-slim image - so the command itself is in a script that runs with "& disown" so it doesn't have a running tty.

**Comment by smijolovic** [comment]
Figured out the issue - figured I would share with the community.

Apt-get has lingering issues with weirdness of stdin when run noninteractive. See here for where these problems continue to exist since 2013:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=728775

In order to address the issue - send stdin to /dev/null for any apt-get command using the -y flag or any other flag that supplies input to apt-get for non-interactive scenarios:

debuerreotype-apt-get buster dist-upgrade -yqq < /dev/null
debuerreotype-apt-get buster install -yqq --no-install-recommends iputils-ping iproute2 < /dev/null
